CONTRIBUTING.md

```diff
@@ -0,0 +1,110 @@
+# Contributing to the Project
+
+This document contains and defines the rules that have to be followed by any
+contributor to the project, in order for any change to be merged into the
+stable branches.
+
+## Workflow
+
+The workflow followed is waterflow. In short, a contributor will target his
+work on the lowest version branch possible starting with 2.0, and the changes
+will automatically be forward-ported to higher versions.
+
+## Restrictions of the development/X.Y branch
+
+The `development/X.Y` branch is a very specific branch in our workflow. No 
+commit must ever go directly into the master branch, and the code shall follow
+one and only one path to get into the master branch.
+
+Merging a pull request into this branch will only ever be done by the
+Bert-E merging bot. Once the necessary approvals and CI passes are met, one 
+only needs to comment `@bert-e approve` to trigger the merging process
+for one's pull request.
+
+## Workflow Guidelines
+
+### Branching Guidelines
+
+When branching, one should choose a name:
+
+* containing a reference to the GitHub issue being tackled
+* starting with a letter, due to a restriction in our merging bot
+* relevant to the goal of the changes made
+
```

Comment:
Mention the valid branch-name prefixes? That can then replace the 2 bullet points above.

Also, maybe provide an example?